router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/marketobulkupload.go

```diff
@@ -0,0 +1,359 @@
+package marketobulkupload
+
+import (
+	"bufio"
+	"encoding/json"
+	stdjson "encoding/json"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	time "time"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	"github.com/rudderlabs/rudder-server/jobsdb"
+	"github.com/rudderlabs/rudder-server/router/batchrouter/asyncdestinationmanager/common"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+type MarketoBulkUploader struct {
+	destName          string
+	destinationConfig map[string]interface{}
+	TransformUrl      string
+	PollUrl           string
+	logger            logger.Logger
+	timeout           time.Duration
+}
+
+func NewManager(destination *backendconfig.DestinationT) (*MarketoBulkUploader, error) {
+	marketoBulkUpload := &MarketoBulkUploader{destName: destination.DestinationDefinition.Name, destinationConfig: destination.Config, PollUrl: "/pollStatus", TransformUrl: config.GetString("DEST_TRANSFORM_URL", "http://localhost:9090")}
+	return marketoBulkUpload, nil
+}
+
+var pkgLogger logger.Logger
+
+func init() {
+	pkgLogger = logger.NewLogger().Child("asyncdestinationmanager").Child("marketobulkupload")
+}
+
+func (b *MarketoBulkUploader) Poll(pollInput common.AsyncPoll) common.PollStatusResponse {
+	payload, err := json.Marshal(pollInput)
+	if err != nil {
+		// needs to be retried
+		b.logger.Error("Error in Marshalling Poll Input: %w", err)
```

Comment:
Can you help me figure out where is the `b.logger ` initialised? I don't see it set in the struct anywhere?

Reply:
initialised